packages/mongodb-dialects/spring-criteria/src/test/kotlin/com/mongodb/jbplugin/dialects/springcriteria/reactive/ReactiveAggregationStagesParserTest.kt

```diff
@@ -0,0 +1,212 @@
+
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import com.mongodb.jbplugin.dialects.springcriteria.IntegrationTest
+import com.mongodb.jbplugin.dialects.springcriteria.ParsingTest
+import com.mongodb.jbplugin.dialects.springcriteria.SpringCriteriaDialectParser
+import com.mongodb.jbplugin.dialects.springcriteria.assert
+import com.mongodb.jbplugin.dialects.springcriteria.collection
+import com.mongodb.jbplugin.dialects.springcriteria.component
+import com.mongodb.jbplugin.dialects.springcriteria.getQueryAtMethod
+import com.mongodb.jbplugin.mql.components.HasAggregation
+import com.mongodb.jbplugin.mql.components.HasCollectionReference
+import com.mongodb.jbplugin.mql.components.HasSourceDialect
+import com.mongodb.jbplugin.mql.components.IsCommand
+import org.junit.jupiter.api.Assertions.assertEquals
+
+@IntegrationTest
+class ReactiveAggregationStagesParserTest {
+    @ParsingTest(
+        fileName = "Book.java",
+        """
+import org.springframework.data.mongodb.core.ReactiveMongoTemplate;
+import org.springframework.data.mongodb.core.aggregation.Aggregation;
+import org.springframework.data.mongodb.core.mapping.Document;
+import java.util.concurrent.Flow;
+
+@Document
+record Book() {}
+
+class Repository {
+    private final ReactiveMongoTemplate template;
+    
+    public Repository(ReactiveMongoTemplate template) {
+        this.template = template;
+    }
+    
+    public Flow.Publisher<Book> allReleasedBooks() {
+        return template.aggregate(
+            Aggregation.newAggregation(),
+            Book.class,
+            Book.class
+        );
+    }
+}
+        """
+    )
+    fun `should be able to parse an empty aggregation using aggregate call with a class type provided for target collection`(
+        psiFile: PsiFile
+    ) {
+        val query = psiFile.getQueryAtMethod("Repository", "allReleasedBooks")
+        SpringCriteriaDialectParser.parse(query).assert(IsCommand.CommandType.AGGREGATE) {
+            component<HasSourceDialect> {
+                assertEquals(HasSourceDialect.DialectName.SPRING_CRITERIA, name)
+            }
+
+            collection<HasCollectionReference.OnlyCollection<PsiElement>> {
+                assertEquals("book", collection)
+            }
+
+            component<HasAggregation<PsiElement>> {
+                assertEquals(0, children.size)
+            }
+        }
+    }
+
+    @ParsingTest(
+        fileName = "Book.java",
+        """
+import org.springframework.data.mongodb.core.ReactiveMongoTemplate;
+import org.springframework.data.mongodb.core.aggregation.Aggregation;
+import org.springframework.data.mongodb.core.mapping.Document;
+import java.util.concurrent.Flow;
+
+@Document
+record Book() {}
+
+class Repository {
+    private final ReactiveMongoTemplate template;
+    
+    public Repository(ReactiveMongoTemplate template) {
+        this.template = template;
+    }
+    
+    public AggregationResults<Book> allReleasedBooks() {
```

Comment:
```suggestion
    public Flow.Publisher<Book> allReleasedBooks() {
```
how important is the return type here? I think that the parsers can still parse correctly without having to specify correct return type still I wonder.

Reply:
You are correct that for our parser is not very relevant but wanted to change the code to make it as correct as possible, so if at some point becomes relevant we don't need to fix 100 tests.

Reply:
updated the rest as well